images/builder/main.go

```diff
@@ -257,30 +280,69 @@ func mergeMaps(maps ...map[string]string) map[string]string {
 
 func parseFlags() options {
 	o := options{}
+	flag.StringVar(&o.buildDir, "build-dir", "", "If provided, this directory will be uploaded as the source for the Google Cloud Build run.")
+	flag.StringVar(&o.cloudbuildFile, "gcb-config", "cloudbuild.yaml", "If provided, this will be used as the name of the Google Cloud Build config file.")
 	flag.StringVar(&o.logDir, "log-dir", "", "If provided, build logs will be sent to files in this directory instead of to stdout/stderr.")
 	flag.StringVar(&o.scratchBucket, "scratch-bucket", "", "The complete GCS path for Cloud Build to store scratch files (sources, logs).")
 	flag.StringVar(&o.project, "project", "", "If specified, use a non-default GCP project.")
 	flag.BoolVar(&o.allowDirty, "allow-dirty", false, "If true, allow pushing dirty builds.")
+	flag.BoolVar(&o.noSource, "no-source", false, "If true, no source will be uploaded with this build.")
 	flag.StringVar(&o.variant, "variant", "", "If specified, build only the given variant. An error if no variants are defined.")
-	flag.StringVar(&o.envPassthrough, "env-passthrough", "", "Comma-separated list of specified environment variables to be passed to GCB as subtitutions with an _ prefix. If the variable doesn't exist, the substitution will exist but be empty.")
+	flag.StringVar(&o.envPassthrough, "env-passthrough", "", "Comma-separated list of specified environment variables to be passed to GCB as substitutions with an _ prefix. If the variable doesn't exist, the substitution will exist but be empty.")
+
 	flag.Parse()
+
 	if flag.NArg() < 1 {
-		_, _ = fmt.Fprintln(os.Stderr, "expected an image directory to be provided")
+		_, _ = fmt.Fprintln(os.Stderr, "expected a config directory to be provided")
 		os.Exit(1)
 	}
-	o.imageDirectory = flag.Arg(0)
+
+	o.configDir = strings.TrimSuffix(flag.Arg(0), "/")
```

Comment:
Why is this change necessary?

Reply:
@Katharine -- I don't think it's strictly necessary, but it's defensive against any weirdness with path joins, double `/`s, etc.

Reply:
`filepath.Clean` takes care of that and more: https://golang.org/pkg/path/filepath/#Clean